.github/workflows/release.yml

```diff
@@ -0,0 +1,75 @@
+name: Release
+
+concurrency:
+  group: ${{ github.workflow }}
+
+on:
+  schedule:
+    - cron: '0 4 * * *'
+  workflow_dispatch:
+    inputs:
+      release-type:
+        description: 'Release type'
+        required: true
+        default: alpha
+        type: choice
+        options:
+          - alpha
+          - stable
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-22.04
+    needs:
+      - lint-and-test
+    steps:
+      - name: Git clone repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          # A custom token must be used, as GitHub doesn't trigger workflows
+          # for pushes made using the default token
+          token: ${{ secrets.RELEASE_BOT_GITHUB_TOKEN }}
+
+      - name: Set up environment
+        uses: ./.github/actions/prepare-environment
+
+      - name: Configure Git user
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+          git config push.autoSetupRemote true
+
+      - name: Prepare release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npm run release:${{ inputs.release-type }}
+
+      - id: git-branch
+        name: Get Git branch
+        run: echo "git-branch=$(git rev-parse --abbrev-ref HEAD)" >> "$GITHUB_OUTPUT"
+
+      - name: Wait for tests to succeed
+        uses: lewagon/wait-on-check-action@3a563271c3f8d1611ed7352809303617ee7e54ac
```

Comment:
Currently this action triggers some deprecation warnings. It was updated in October though so hopefully they get fixed soon (there are some PRs open). Have pinned it to the hash as it's a third-party action (recommended for security).